stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -193,11 +219,19 @@ extension Unsafe${Mutable}BufferPointer: ${Mutable}Collection, RandomAccessColle
     // optimizer is not capable of creating partial specializations yet.
     // NOTE: Range checks are not performed here, because it is done later by
     // the subscript function.
-    let l = limit - i
+    // NOTE: Unchecked math because unsafe pointers are not required to
+    // behave sensibly if you misuse their indices. We validate in
+    // debug mode.
+    let result = limit.subtractingReportingOverflow(i)
```

Comment:
The name `result` isn't a great choice here -- this variable doesn't hold the result of this function.